app/controllers/application_controller.rb

```diff
@@ -203,6 +203,10 @@ def bust_content_change_caches
     Settings::General.admin_action_taken_at = Time.current # Used as cache key
   end
 
+  def set_current_user
+    @current_user = current_user
```

Comment:
@rhymes Yeah, of course, my first attempt was that, without this extra setup 😊. The thing is all the requests where we need this new `current_user` filter for hiding/showing/collapsing comments, are prepended with `set_cache_control_headers`. And there is a concern https://github.com/forem/forem/blob/3fb69eca20ab6e6a995b1768de29d739aea56f5f/app/controllers/concerns/edge_cache_safety_check.rb in play that doesn't let one use `current_user` anywhere down the tree in that request 😓. It throws this error if we attempt using the `current_user` directly in such requests: 
```
You may not use current_user in this cached code path.
```
I know this doesn't look good, but this was the only way I could make `current_user` work in these requests, by setting an instance variable `@current_user` (the name can change) to it **before** `set_cache_control_headers` is called, for each of these requests.

Suggestion welcome. Happy to change it to something better. 🙏

Reply:
I believe @rhymes is correct

> the workaround would yield the same behavior though on the edge

The proposed solution here seems to work around the safety check put in place, leading to the same issues the check is trying to help avoid. These endpoints are cached so we can't refer to `current_user`. If/when we do, the `current_user` is cached so it basically becomes the content for whatever `current_user` hits the page first, warming up the cache. Then, the following request for the same content by a different user shows the wrong data because the wrong user data is cached.

We may have to move more of this logic/code to the client-side in JS where the content isn't cached. There are references throughout the codebase to getting the "current user". One example:

https://github.com/forem/forem/blob/f607cf8722988ed09bb3eae96c332fb1a612205b/app/javascript/packs/profileDropdown.js#L18

Reply:
@atsmith813 Thanks a lot for the explanation. 👍

@aitchiss @rhymes  I'll change the approach and try coming up with a client-side solution. Most of the current code in this PR will be gone. Will keep you folks posted here. Thanks for the support 👍

Reply:
Thank you for being so forthcoming on feedback @akashdotsrivastava!